### PR TITLE
IBM MQ test connection tool

### PR DESCRIPTION
**Pull request by zjingjbj**
We will create a troubleshooting doc for IBM MQ sensor of the instana product. In the doc, there is a step to use a test jar to try the IBM MQ connection. The test jar is put in this git hub, together with the usage.

### CODE REVIEW

File: ibmmq/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+2. Client binding mode
+With this mode, you need to provide the channel name and channel port to try the IBM MQ connection. If the IBM MQ channel is security enabled, username and password is also needed. If you try this mode, you can run this jar on other server or the same server that the IBM MQ is running. 
+
+## Usage
+
+You can get the whole usge with the command:
+java -jar ./testMQ.jar 
+
+The output is like:
+If you want to try local binding. The usage is:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:17]:
```suggestion
The usage output of local binding is as follows: 
```

File: ibmmq/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+With this mode, you need to provide the channel name and channel port to try the IBM MQ connection. If the IBM MQ channel is security enabled, username and password is also needed. If you try this mode, you can run this jar on other server or the same server that the IBM MQ is running. 
+
+## Usage
+
+You can get the whole usge with the command:
+java -jar ./testMQ.jar 
+
+The output is like:
+If you want to try local binding. The usage is: 
+    java -jar ./testMQ.jar -m <qmgr-name> -a <lib-path> [-q <queueName>]
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:18]:
```suggestion
   ```
    java -jar ./testMQ.jar -m <qmgr-name> -a <lib-path> [-q <queueName>]
```
{: codeblock}
```

File: ibmmq/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+If you want to try client binding. The usage is:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:20]:
```suggestion
The usage output of client binding mode is as follows:
```

File: ibmmq/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    java -jar ./testMQ.jar -h <host> -p <port> -c <channel> [-u <user>] [-z <password>] [-q <queueName>] [-k <keystore>] [-w <keystore-password>] -s [<ciph-suite>]
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:21]:
``` 
java -jar ./testMQ.jar -h <host> -p <port> -c <channel> [-u <user>] [-z <password>] [-q <queueName>] [-k <keystore>] [-w <keystore-password>] -s [<ciph-suite>]
```
  {: codeblock}

File: ibmmq/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Here are the examples:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:23]:
```suggestion
You can refer following examples to view the usage:
```

File: ibmmq/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+1. Local binding mode:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:25]:
```suggestion
- Local binding mode:
```

File: ibmmq/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+java -jar ./testMQ.jar -m qmName -a /opt/mqm/java/lib64 -q SYSTEM.ADMIN.ACCOUNTING.QUEUE
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:26]:
```
java -jar ./testMQ.jar -m qmName -a /opt/mqm/java/lib64 -q SYSTEM.ADMIN.ACCOUNTING.QUEUE
```
{: codeblock}

File: ibmmq/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+2. Client binding mode:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:28]:
```suggestion
- Client binding mode:
```

File: ibmmq/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+java -jar ./testMQ.jar -m qmName -h 1.2.3.4 -p 1801 -c SYSTEM.AUTO.SVRCONN -u root -z dummyPwd -q AAA
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:29]:
```
java -jar ./testMQ.jar -m qmName -h 1.2.3.4 -p 1801 -c SYSTEM.AUTO.SVRCONN -u root -z dummyPwd -q AAA
```
{: codeblock}

File: ibmmq/README.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+2. Client binding mode
+With this mode, you need to provide the channel name and channel port to try the IBM MQ connection. If the IBM MQ channel is security enabled, username and password is also needed. If you try this mode, you can run this jar on other server or the same server that the IBM MQ is running. 
+
+## Usage
+
+You can get the whole usge with the command:
+java -jar ./testMQ.jar 
+
+The output is like:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:16]:
```suggestion
```

File: ibmmq/README.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+1. Local binding mode
+With this mode, the only parameter you need to provide is the IBM MQ lib path when you want to try the IBM MQ connection. If you try this mode, you need to run this jar on the same server where the IBM MQ is running.
+
+2. Client binding mode
+With this mode, you need to provide the channel name and channel port to try the IBM MQ connection. If the IBM MQ channel is security enabled, username and password is also needed. If you try this mode, you can run this jar on other server or the same server that the IBM MQ is running. 
+
+## Usage
+
+You can get the whole usge with the command:
+java -jar ./testMQ.jar
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:14]:
```suggestion
   ```bash
   java -jar ./testMQ.jar
```
  {: codeblock}
```

File: ibmmq/README.md (hunk 2 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:18]:
```suggestion
   ```bash
    java -jar ./testMQ.jar -m <qmgr-name> -a <lib-path> [-q <queueName>]
```
{: codeblock}
```

File: ibmmq/README.md (hunk 4 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:21]:
```suggestion
```bash
java -jar ./testMQ.jar -h <host> -p <port> -c <channel> [-u <user>] [-z <password>] [-q <queueName>] [-k <keystore>] [-w <keystore-password>] -s [<ciph-suite>]
```
  {: codeblock}
```

File: ibmmq/README.md (hunk 7 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:26]:
```suggestion
```bash
java -jar ./testMQ.jar -m qmName -a /opt/mqm/java/lib64 -q SYSTEM.ADMIN.ACCOUNTING.QUEUE
```
{: codeblock}
```

File: ibmmq/README.md (hunk 9 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:29]:
```suggestion
```bash
java -jar ./testMQ.jar -m qmName -h 1.2.3.4 -p 1801 -c SYSTEM.AUTO.SVRCONN -u root -z dummyPwd -q AAA
```
{: codeblock}
```

**Comment by Subhalakshmi-Moosad** [comment]
Sent my suggestions to Koustav through Slack.

File: ibmmq/README.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+1. Local binding mode
+With this mode, the only parameter you need to provide is the IBM MQ lib path when you want to try the IBM MQ connection. If you try this mode, you need to run this jar on the same server where the IBM MQ is running.
+
+2. Client binding mode
+With this mode, you need to provide the channel name and channel port to try the IBM MQ connection. If the IBM MQ channel is security enabled, username and password is also needed. If you try this mode, you can run this jar on other server or the same server that the IBM MQ is running. 
+
+## Usage
+
+You can get the whole usge with the command:
```
[the remark this answers is not in the document]
Comment (koustav-mukherjee) [reply ibmmq/README.md:13]: What do mean by whole usage?

**Review by koustav-mukherjee — COMMENTED** [review]
done

**Comment by meghasalunke1** [comment]
@koustav-mukherjee @Subhalakshmi-Moosad I've some more comments, will update Koustav. Thanks!

@zjingjbj Please don't incorporate our suggestions as we're still reviewing the file. Thanks!

File: ibmmq/README.md (hunk 9 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:29]:
```suggestion
```

**Review by koustav-mukherjee — COMMENTED** [review]
done

File: ibmmq/README.md (hunk 11 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:14]:
```suggestion
```

File: ibmmq/README.md (hunk 2 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:18]:
```suggestion
```

File: ibmmq/README.md (hunk 4 above)
Comment (koustav-mukherjee) [inline ibmmq/README.md:21]:
```suggestion
```

**Review by koustav-mukherjee — COMMENTED** [review]
done

**Comment by meghasalunke1** [comment]
Shared my suggestions to Koustav and Subha. Thanks!

File: ibmmq/README.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+# IBM MQ test jar
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:1]:
```suggestion
# IBM MQ test JAR
```

**Review by koustav-mukherjee — COMMENTED** [review]
applied comment

**Review by koustav-mukherjee — COMMENTED** [review]
Done

**Review by koustav-mukherjee — COMMENTED** [review]
Applied

**Comment by Subhalakshmi-Moosad** [comment]
@zjingjbj Please incorporate Koustav's comments. Thanks. Also, update the commands using codeblock. Thanks.

**Comment by koustav-mukherjee** [comment]
@zjingjbj , could you please check when can you incorporate the comments and close the PR.

File: ibmmq/README.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+The purpose of the IBM MQ connection test tool is to test whether IBM MQ is configured and can be connected. You can select any of the following connection modes:
+
+1. Local binding mode: Provide the `IBM MQ lib` path parameter. You can run the IBM MQ test tool on the same server where the IBM MQ runs.
+
+2. Client binding mode: Provide the channel name and channel port to use the IBM MQ connection. If the security is enabled for the IBM MQ channel, then you need a username and password to log in to the channel. By using client binding mode, you can run the IBM MQ test tool on another server or on the same server where the IBM MQ runs. 
+
+## Usage of the IBM MQ connection test tool
+
+- You need to export JAVA_HOME before the tool is invoked.
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:11]:
```suggestion
- You need to export JAVA_HOME before you use the IBM MQ connection test tool.
```

File: ibmmq/README.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+- You can get the whole usage with the command: ./testMQ.sh -help
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:13]:
```suggestion
- You can obtain the whole usage with the command: ./testMQ.sh -help 
```

File: ibmmq/README.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+2. Client binding mode: Provide the channel name and channel port to use the IBM MQ connection. If the security is enabled for the IBM MQ channel, then you need a username and password to log in to the channel. By using client binding mode, you can run the IBM MQ test tool on another server or on the same server where the IBM MQ runs. 
+
+## Usage of the IBM MQ connection test tool
+
+- You need to export JAVA_HOME before the tool is invoked.
+
+- You can get the whole usage with the command: ./testMQ.sh -help 
+
+- To use the IBM MQ connection test tool in the local binding mode, run the following sample command:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:15]: Add an empty line after line no 15

File: ibmmq/README.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+- You need to export JAVA_HOME before the tool is invoked.
+
+- You can get the whole usage with the command: ./testMQ.sh -help 
+
+- To use the IBM MQ connection test tool in the local binding mode, run the following sample command: 
+./testMQ.sh -m <qmgr-name> -a <lib-path> [-q <queueName>]
+
+For example, ./testMQ.sh -m qmName -a /opt/mqm/java/lib64 -q SYSTEM.ADMIN.ACCOUNTING.QUEUE
+
+- To use the IBM MQ connection test tool in the client binding mode, run the following sample command:
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:20]: Add an empty line after line no 20

**Review by koustav-mukherjee — COMMENTED** [review]
done

File: ibmmq/README.md (hunk 15 above)
Comment (zjingjbj) [reply ibmmq/README.md:13]: I mean the whole parameter's will be printed out.

Comment (zjingjbj) [reply ibmmq/README.md:13]: I'll remove the word `whole` if you like.

File: ibmmq/testMQ.sh [hunk 18]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (zjingjbj) [reply ibmmq/testMQ.sh:1]: This script is used to invoke testMQ.jar, which is described in README.md. So I think we don't need to write comment for this.

File: ibmmq/README.md (hunk 14 above)
Comment (zjingjbj) [reply ibmmq/README.md:11]: `JAVA_HOME` is an environment variable. It's not a file. So I'll modify this sentence as
```suggestion
- You need to export the `JAVA_HOME` before you use the IBM MQ connection test tool.
```

File: ibmmq/README.md (hunk 15 above)
Comment (zjingjbj) [reply ibmmq/README.md:13]:
```suggestion
- You can obtain the usage with the command: ./testMQ.sh -help 
```

**Comment by zjingjbj** [comment]
@koustav-mukherjee I've updated according to your comments. Can you help review? Thanks!

File: ibmmq/README.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+1. Local binding mode: Provide the `IBM MQ lib` path parameter. You can run the IBM MQ test tool on the same server where the IBM MQ runs.
+
+2. Client binding mode: Provide the channel name and channel port to use the IBM MQ connection. If the security is enabled for the IBM MQ channel, then you need a username and password to log in to the channel. By using client binding mode, you can run the IBM MQ test tool on another server or on the same server where the IBM MQ runs. 
+
+## Usage of the IBM MQ connection test tool
+
+- You need to export the `JAVA_HOME` before you use the IBM MQ connection test tool.
+
+- You can obtain the usage with the command: ./testMQ.sh -help
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:13]: can we write **./testMQ.sh -help** in codeblock?

File: ibmmq/README.md [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+
+The purpose of the IBM MQ connection test tool is to test whether IBM MQ is configured and can be connected. You can select either of the following connection modes:
+
+1. Local binding mode: Provide the `IBM MQ lib` path parameter. You can run the IBM MQ test tool on the same server where the IBM MQ runs.
+
+2. Client binding mode: Provide the channel name and channel port to use the IBM MQ connection. If the security is enabled for the IBM MQ channel, then you need a username and password to log in to the channel. By using client binding mode, you can run the IBM MQ test tool on another server or on the same server where the IBM MQ runs. 
+
+## Usage of the IBM MQ connection test tool
+
+- You need to export the `JAVA_HOME` before you use the IBM MQ connection test tool.
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:11]: If `JAVA_HOME` is a file then pls mention word file after `JAVA_HOME`.

File: ibmmq/README.md [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+./testMQ.sh -m <qmgr-name> -a <lib-path> [-q <queueName>]
+```
+
+For example, ./testMQ.sh -m qmName -a /opt/mqm/java/lib64 -q SYSTEM.ADMIN.ACCOUNTING.QUEUE
+
+- To use the IBM MQ connection test tool in the client binding mode, run the following sample command: 
+
+```
+./testMQ.sh -m <qmgr-name> -h <host> -p <port> -c <channel> [-u <user>] [-z <password>] [-q <queueName>] [-k <keystore>] [-w <keystore-password>] -s [<ciph-suite>]
+```
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:27]: Please write {: codeblock} after closing ```

File: ibmmq/README.md [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+
+- You need to export the `JAVA_HOME` before you use the IBM MQ connection test tool.
+
+- You can obtain the usage with the command: ./testMQ.sh -help 
+
+- To use the IBM MQ connection test tool in the local binding mode, run the following sample command: 
+
+```
+./testMQ.sh -m <qmgr-name> -a <lib-path> [-q <queueName>]
+```
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:19]: Please write {: codeblock} after closing ```

File: ibmmq/README.md (hunk 20 above)
Comment (zjingjbj) [reply ibmmq/README.md:11]: `JAVA_HOME` isn't a file but an environment variable.

File: ibmmq/README.md (hunk 19 above)
Comment (zjingjbj) [reply ibmmq/README.md:13]: Yes, sure.

**Comment by zjingjbj** [comment]
@koustav-mukherjee I've modified. Can you help review? Thanks!

File: ibmmq/README.md (hunk 20 above)
Comment (koustav-mukherjee) [reply ibmmq/README.md:11]: Ok, then please write 'You need to export the `JAVA_HOME` environment variable before you use the IBM MQ connection test tool.'

File: ibmmq/README.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+
+The purpose of the IBM MQ connection test tool is to test whether IBM MQ is configured and can be connected. You can select either of the following connection modes:
+
+1. Local binding mode: Provide the `IBM MQ lib` path parameter. You can run the IBM MQ test tool on the same server where the IBM MQ runs.
+
+2. Client binding mode: Provide the channel name and channel port to use the IBM MQ connection. If the security is enabled for the IBM MQ channel, then you need a username and password to log in to the channel. By using client binding mode, you can run the IBM MQ test tool on another server or on the same server where the IBM MQ runs. 
+
+## Usage of the IBM MQ connection test tool
+
+- You need to export the `JAVA_HOME` before you use the IBM MQ connection test tool.
```
Comment (koustav-mukherjee) [inline ibmmq/README.md:11]:
```suggestion
- You need to export the `JAVA_HOME` environment variable before you use the IBM MQ connection test tool.
```

**Comment by zjingjbj** [comment]
@koustav-mukherjee Can you help review as I've completed the changes? Thanks!

**Review by koustav-mukherjee — APPROVED** [review]

**Pull request merged**